.github/workflows/build.yaml

```diff
@@ -0,0 +1,29 @@
+name: Check build
+on:
+  push:
+    branches: [main]
```

Comment:
I am not sure why we should run this on `push` event.
There is a build image on `push` to main.
When this will be useful?

Reply:
1. It runs the tests so we can always be sure that the main branch passes our test suite (once we have more tests).
2. It will probably be quicker most of the time.
3. It's good practice to run the same checks for PRs as well as pushes.